Docs: Explain non-language directories in top-level README.md (#4921)
awsdocs/aws-doc-sdk-examples: README.md

Within each directory, you will find SDK-specific instructions for understanding and invoking example code.

### Additional directories

| directory                     | purpose                                                                                                                                                     | usage                                                                                                                                                                               |
|-------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | [/applications](applications) | Contains the non-language-specific components of example applications, which show how the SDKs can be used in the context of a production-like application. | To view the language-specific components for each example application, see the `cross-service` folder in the sub-directory for your desired language (e.g. `python/cross-service`). |

Reviewer: ```suggestion
| [/applications](applications) | Contains the non-language-specific components of example applications, which show how the SDKs can be used in the context of a production-like application. | To view the language-specific components for each example application, see the `cross-service` folder in the sub-directory for your desired language (such as `python/cross-service`). |
```

> | [/test](test)                 | Contains all components supporting the custom test automation framework used to routinely test the code in this repository.                                 | Deploys to AWS as polyglot container-based integration testing solution. WARNING: Still under active construction as of 2023.                                                       |

Reviewer: ```suggestion
| [/test](test)                 | Contains all components supporting the custom test automation framework used to routinely test the code in this repository.                                 | Deploys to AWS as a polyglot container-based integration testing solution. WARNING: Still under active construction as of 2023.                                                       |
```

> | [/resources](resources)       | Contains shared components used by many code examples across this repository.                                                                               | Deploys as front-end ([/clients](/resources/clients)) or back-end ([/cdk](/resources/cdk) or [/cfn](/resources/cfn)) components.                                                                                                                                                                      

Reviewer: ```suggestion
| [/resources](resources)       | Contains shared components used by many code examples across this repository.                                                                               | Deploys as frontend ([/clients](/resources/clients)) or backend ([/cdk](/resources/cdk) or [/cfn](/resources/cfn)) components.                                                                                                                                                                      
```
